modules/kummaBidAdapter.js

```diff
@@ -1,63 +1,125 @@
-var CONSTANTS = require('src/constants.json');
-var adaptermanager = require('src/adaptermanager');
-
-var KummaAdapter = function KummaAdapter() {
-  function _callBids(params) {
-    var bidURL;
-    var bids = params.bids || [];
-    var requestURL = window.location.protocol + '//cdn.kumma.com/pb_ortb.js?cb=' + new Date().getTime() + '&ver=1&';
-
-    for (var i = 0; i < bids.length; i++) {
-      var requestParams = {};
-      var bid = bids[i];
-
-      requestParams.pub_id = bid.params.pubId;
-      requestParams.site_id = bid.params.siteId;
-      requestParams.placement_id = bid.placementCode;
-
-      var parseSized = utils.parseSizesInput(bid.sizes);
-      var arrSize = parseSized[0].split('x');
-
-      requestParams.width = arrSize[0];
-      requestParams.height = arrSize[1];
-      requestParams.callback = '$$PREBID_GLOBAL$$._doKummaCallback';
-      requestParams.callback_uid = bid.bidId;
-      bidURL = requestURL + utils.parseQueryStringParameters(requestParams);
-
-      utils.logMessage('Kumma.prebid, Bid ID: ' + bid.bidId + ', Pub ID: ' + bid.params.pubId);
-      adloader.loadScript(bidURL);
-    }
-  }
-
-  $$PREBID_GLOBAL$$._doKummaCallback = function (response) {
-    var bidObject;
-    var bidRequest;
-    var callbackID;
-    callbackID = response.callback_uid;
-    bidRequest = utils.getBidRequest(callbackID);
-    if (response.cpm > 0) {
-      bidObject = bidfactory.createBid(CONSTANTS.STATUS.GOOD, bidRequest);
-      bidObject.bidderCode = 'kumma';
-      bidObject.cpm = response.cpm;
-      bidObject.ad = response.tag;
-      bidObject.width = response.width;
-      bidObject.height = response.height;
-    } else {
-      bidObject = bidfactory.createBid(CONSTANTS.STATUS.NO_BID, bidRequest);
-      bidObject.bidderCode = 'kumma';
-      utils.logMessage('No Bid response from Kumma request: ' + callbackID);
-    }
-    bidmanager.addBidResponse(bidRequest.placementCode, bidObject);
-  };
-
-  return {
-    callBids: _callBids
-  };
-};
-adaptermanager.registerBidAdapter(new KummaAdapter(), 'kumma');
-
-module.exports = KummaAdapter;
+
+import {logError, getTopWindowLocation} from 'src/utils';
+import { registerBidder } from 'src/adapters/bidderFactory';
+
+export const spec = {
+
+  code: 'kumma',
+
+  isBidRequestValid: bid => (
+    !!(bid && bid.params && bid.params.pubId && bid.params.siteId)
+  ),
+  buildRequests: bidRequests => {
+    const request = {
+      id: bidRequests[0].bidderRequestId,
+      at: 2,
+      imp: bidRequests.map(slot => impression(slot)),
+      site: site(bidRequests),
+      device: device(),
+    };
+    return {
+      method: 'POST',
+      url: '//hb.kumma.com/',
+      data: JSON.stringify(request),
+    };
+  },
+  interpretResponse: (response, request) => (
+    bidResponseAvailable(request, response)
+  ),
+};
+function bidResponseAvailable(bidRequest, bidResponse) {
+  const idToImpMap = {};
+  const idToBidMap = {};
+  const ortbRequest = parse(bidRequest.data);
```

Comment:
You can simplify this quite a bit by just throwing the error here. The `bidderFactory` runs `interpretResponse` in a try/catch, and does a `logError` if anything goes wrong.

Also... your `parse` function just returns `null` on error, which will make `openrtbRequest.imp` throw an error anyway.